server/services/auth.js

```diff
@@ -0,0 +1,52 @@
+import jwt from 'jsonwebtoken';
+
+import config from '../../config/env';
+import User from '../models/user';
+import userService from '../services/user';
+
+function loginUser(email) {
+  return new Promise((resolve) => {
+    User.getByEmail(email)
+      .then(() => {
+        const token = jwt.sign({ email }, config.auth.jwt_secret);
+        resolve(token);
+      })
+      .catch((err) => {
+        resolve(err);
+      });
+  });
+}
+
+function signupUser(body) {
+  const userObj = {
+    email: body.email,
+    name: body.name
+  };
+
+  const returnObj = {};
+
+  return new Promise((resolve) => {
+    User.getByEmail(userObj.email)
+      .then(() => {
+        returnObj.error = 'User already exists';
+        resolve(returnObj);
+      })
+      .catch(() => {
+        userService.createUser(userObj)
```

Comment:
Try not to nest promises, instead return the `.createuser()` and chain on the original promise.

```
.then(() => userService.createUser(userObj))
.then((user) => {
  ...
})
```